### PR TITLE
ESYS: Related to GetNonceTPM

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Fixes: #1020 
Fixes: #1021

### CODE REVIEW

File: src/tss2-esys/esys_tr.c [hunk 1]
```diff
@@ -509,3 +509,48 @@ Esys_TRSess_SetAttributes(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
...
 }
+
+/** Retrieve the TPM nonce of an Esys_TR session object.
+ *
+ * Some operations (i.e. Esys_PolicySigned) require the nonce returned by the
+ * TPM during Esys_StartauthSession. This function provides this nonce to the
+ * caller.
+ * @param esys_context [in,out] The ESYS_CONTEXT.
+ * @param esys_handle [in,out] The ESYS_TRsess for which to retrieve the nonce.
+ * @param nonceTPM [out] The nonce of the object (callee-allocated; use free()).
```
Comment (tstruk) [inline src/tss2-esys/esys_tr.c]: s/use free/user free/

File: src/tss2-esys/esys_tr.c [hunk 2]
```diff
@@ -509,3 +509,48 @@ Esys_TRSess_SetAttributes(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
...
+        r = TSS2_ESYS_RC_BAD_TR;
+        goto_if_error(r, "NonceTPM for non-session object requested.",
+                      error_cleanup);
+
+    }
+    **nonceTPM = esys_object->rsrc.misc.rsrc_session.nonceTPM;
+
+    return r;
+ error_cleanup:
+    SAFE_FREE(nonceTPM);
```
Comment (tstruk) [inline src/tss2-esys/esys_tr.c]: SAFE_FREE(*nonceTPM);

File: include/tss2/tss2_esys.h [hunk 3]
```diff
@@ -309,8 +309,7 @@ Esys_StartAuthSession(
     TPM2_SE sessionType,
```
Comment (tstruk) [inline include/tss2/tss2_esys.h]: I'm not sure if I'm looking at the right spec, but according to the latest I could find Esys_StartAuthSession() function also takes TPM2B_ENCRYPTED_SECRET const *encryptedSalt,
See https://trustedcomputinggroup.org/wp-content/uploads/TSS_TSS-2.0-Enhanced-System-API_V0.9_R03_Public-Review-1.pdf
section 11.3.6 Esys_StartAuthSession.

Comment (AndreasFuchsSIT) [reply include/tss2/tss2_esys.h]: latest is https://trustedcomputinggroup.org/wp-content/uploads/TSS_ESAPI_Version-0.9_Revision-04_reviewEND030918.pdf

File: src/tss2-esys/esys_tr.c (hunk 1 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_tr.c]: this was actually meant as a verb to use free().

File: src/tss2-esys/esys_tr.c (hunk 2 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_tr.c]: done

**Pull request merged**